packages/@coorpacademy-components/src/molecule/payment-form/index.js

```diff
@@ -0,0 +1,180 @@
+import React, {PropTypes} from 'react';
+import style from './style.css';
+
+class PaymentForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.stripe = props.stripeInstance ? props.stripeInstance(props.stripeKeyPublic) : null;
+    this.state = {
+      cardNumberHasError: false,
+      cardExpiryHasError: false,
+      cardCvcHasError: false,
+      submitButtonEnabled: true
+    };
+  }
+
+  componentDidMount() {
+    if (this.stripe) {
+      const elements = this.stripe.elements();
+
+      const elementOptions = {
+        style: {
+          base: {
+            iconStyle: 'solid',
+            color: '#546E7A',
+            fontSize: '16px',
+            fontFamily: '"Open Sans"',
+            '::placeholder': {
+              color: '#546E7A',
+              fontSize: '16px',
+              fontFamily: '"Open Sans"'
+            }
+          }
+        }
+      };
+
+      this.cardNumber = elements.create('cardNumber', elementOptions);
+      this.cardExpiry = elements.create('cardExpiry', elementOptions);
+      this.cardCvc = elements.create('cardCvc', elementOptions);
+
+      this.cardNumber.mount('#card-number');
+      this.cardExpiry.mount('#card-expiry');
+      this.cardCvc.mount('#card-cvc');
+
+      this.cardNumber.on('change', ({error}) => {
+        this.setState({
+          cardNumberHasError: error,
+          submitButtonEnabled: true
+        });
+      });
+
+      this.cardExpiry.on('change', ({error}) => {
+        this.setState({
+          cardExpiryHasError: error,
+          submitButtonEnabled: true
+        });
+      });
+
+      this.cardCvc.on('change', ({error}) => {
+        this.setState({
+          cardCvcHasError: error,
+          submitButtonEnabled: true
+        });
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    this.cardNumber.unmount();
+    this.cardExpiry.unmount();
+    this.cardCvc.unmount();
+  }
+
+  handleSubmit() {
+    if (this.stripe) {
+      return this.stripe.createToken(this.cardNumber)
+      .then(response => {
+        if (!response.error)
+          this.setState({submitButtonEnabled: false});
+        return this.props.createSubscription(response);
+      });
+    }
+  }
+
+  render() {
+    const {
+      submitText,
+      checkImage,
+      cardsImage,
+      cardOwnerEmail,
+      acceptedCardsText,
+      securedPaymentText,
+      securedPaymentImage,
+      disclaimer,
+      errors,
+      warning
+    } = this.props;
+
+    const {
+      cardNumberHasError,
+      cardExpiryHasError,
+      cardCvcHasError,
+      submitButtonEnabled
+    } = this.state;
+
+    const handleSubmit = () => this.handleSubmit();
```

Comment:
hum on en a besoin pour binder le this, le linter ne laisse pas passer le `.bind`